src/inference/src/core.cpp

```diff
@@ -11,31 +11,46 @@
 #include "openvino/core/so_extension.hpp"
 #include "openvino/runtime/device_id_parser.hpp"
 #include "openvino/runtime/iremote_context.hpp"
+#include "openvino/util/common_util.hpp"
 #include "openvino/util/file_util.hpp"
 
 namespace ov {
 
 std::string find_plugins_xml(const std::string& xml_file) {
```

Comment:
May we use next algorithm?
1. If path exists, return path
2. If path contains file separator, return path and not search in OV folder (it will fail on load by pugixml). Or throw exception as not supported format to be looked in OV folder
3. If path doesn't contain file separator, search it in OV folder
4. If path is empty, use `plugins.xml`, search it in OV folder

It should be safe and not very strict because:
1. User is allowed to set any path. The possible problem is may be in file name collision between CWD/<filename> and OV/<filename> - but all security issues will be user responsibility
2. We search only under secure OV folder with no restriction on file name and extension